vmnet.go

```diff
@@ -0,0 +1,419 @@
+// see: https://developer.apple.com/documentation/vmnet/vmnet_return_t?language=objc
+type VmnetReturn C.uint32_t
+
+const (
+	ErrVmnetSuccess            VmnetReturn = C.VMNET_SUCCESS              // VMNET_SUCCESS Successfully completed.
+	ErrVmnetFailure            VmnetReturn = C.VMNET_FAILURE              // VMNET_FAILURE General failure.
+	ErrVmnetMemFailure         VmnetReturn = C.VMNET_MEM_FAILURE          // VMNET_MEM_FAILURE Memory allocation failure.
+	ErrVmnetInvalidArgument    VmnetReturn = C.VMNET_INVALID_ARGUMENT     // VMNET_INVALID_ARGUMENT Invalid argument specified.
+	ErrVmnetSetupIncomplete    VmnetReturn = C.VMNET_SETUP_INCOMPLETE     // VMNET_SETUP_INCOMPLETE Interface setup is not complete.
+	ErrVmnetInvalidAccess      VmnetReturn = C.VMNET_INVALID_ACCESS       // VMNET_INVALID_ACCESS Permission denied.
+	ErrVmnetPacketTooBig       VmnetReturn = C.VMNET_PACKET_TOO_BIG       // VMNET_PACKET_TOO_BIG Packet size larger than MTU.
+	ErrVmnetBufferExhausted    VmnetReturn = C.VMNET_BUFFER_EXHAUSTED     // VMNET_BUFFER_EXHAUSTED Buffers exhausted in kernel.
+	ErrVmnetTooManyPackets     VmnetReturn = C.VMNET_TOO_MANY_PACKETS     // VMNET_TOO_MANY_PACKETS Packet count exceeds limit.
+	ErrVmnetSharingServiceBusy VmnetReturn = C.VMNET_SHARING_SERVICE_BUSY // VMNET_SHARING_SERVICE_BUSY Vmnet Interface cannot be started as conflicting sharing service is in use.
+	ErrVmnetNotAuthorized      VmnetReturn = C.VMNET_NOT_AUTHORIZED       // VMNET_NOT_AUTHORIZED The operation could not be completed due to missing authorization.
+)
+
+var _ error = VmnetReturn(0)
+
+func (e VmnetReturn) Error() string {
+	switch e {
+	case ErrVmnetSuccess:
+		return "Vmnet: Successfully completed"
+	case ErrVmnetFailure:
+		return "Vmnet: Failure"
+	case ErrVmnetMemFailure:
+		return "Vmnet: Memory allocation failure"
+	case ErrVmnetInvalidArgument:
+		return "Vmnet: Invalid argument specified"
+	case ErrVmnetSetupIncomplete:
+		return "Vmnet: Interface setup is not complete"
+	case ErrVmnetInvalidAccess:
+		return "Vmnet: Permission denied"
+	case ErrVmnetPacketTooBig:
+		return "Vmnet: Packet size larger than MTU"
+	case ErrVmnetBufferExhausted:
+		return "Vmnet: Buffers exhausted in kernel"
+	case ErrVmnetTooManyPackets:
+		return "Vmnet: Packet count exceeds limit"
+	case ErrVmnetSharingServiceBusy:
+		return "Vmnet: Vmnet Interface cannot be started as conflicting sharing service is in use"
+	case ErrVmnetNotAuthorized:
+		return "Vmnet: The operation could not be completed due to missing authorization"
+	default:
+		return fmt.Sprintf("Vmnet: Unknown error %d", uint32(e))
+	}
+}
+
+// VmnetMode defines the mode of a vmnet network.
+// see: https://developer.apple.com/documentation/vmnet/operating_modes_t?language=objc
+type VmnetMode uint32
+
+const (
+	HostMode   VmnetMode = C.VMNET_HOST_MODE
+	SharedMode VmnetMode = C.VMNET_SHARED_MODE
+	// Deprecated: BridgedMode is not supported by NewVmnetNetworkConfiguration
```

Comment:
Do you know why not supported?

Reply:
I don't know. It's documented on:
https://developer.apple.com/documentation/vmnet/vmnet_network_configuration_create(_:_:)?language=objc
> Parameters
  mode
    Shared mode or host-only mode.

Reply:
Why BridgedMode is Deprecated? Can you link to the docs about this?

Since it is not supported we should not include it, marking it as Deprecated look wrong.